### PR TITLE
Implement redirects from Whitehall finder paths

### PR DESCRIPTION
**Pull request by brucebolt**
These redirects used to be rendered by Whitehall, but were removed in https://github.com/alphagov/whitehall/pull/6950.

As Whitehall Frontend is being retired, we need to bring these into another application.  The previous routes are already being redirected to Finder Frontend (`/search/all` for publications and `/search/news-and-communications` for announcements), so are best processed here.

This replicates the behaviour in an identical way to Whitehall, the commit messages link to the redirect code before it was removed.

[Trello card](https://trello.com/c/Drb1H01H/383-bug-fix-redirects-from-whitehall-rendered-finders), [Jira issue PP-892](https://gov-uk.atlassian.net/browse/PP-892)

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

Follow [these steps](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) if you are doing a Rails upgrade.

### CODE REVIEW

File: app/controllers/finders_controller.rb [hunk 1]
```diff
@@ -196,4 +198,143 @@ def include_ecommerce?
       /government/organisations/hm-revenue-customs/contact
     ].exclude?(content_item.base_path)
   end
+
+  def identify_and_redirect_legacy_params
```
Comment (BeckaL) [inline app/controllers/finders_controller.rb:202]: Can we just in-line this method? Seems readable enough to have this in the body

File: app/controllers/finders_controller.rb [hunk 2]
```diff
@@ -196,4 +198,143 @@ def include_ecommerce?
...
+      subtaxons
+      taxons
+      to_date
+    ].freeze
+    given_params = params.keys.map(&:to_sym)
+
+    (legacy_params & given_params).any?
+  end
+
+  def redirect_legacy_params
```
Comment (BeckaL) [inline app/controllers/finders_controller.rb]: Naming suggestion - `redirect_transforming_legacy_params`? Maybe just me, but when I read this method I assume it's returning me some params for a redirect, not actually doing the redirect itself.

Comment (BeckaL) [reply app/controllers/finders_controller.rb]: Also, do you think there's a sensible way of splitting this method up into slightly smaller ones? Find it slightly hard to parse this. Edit: Ah, another thing I should have waited to read the next commit for! Maybe you could move the refactor splitting out the taxons stuff into this commit?

File: app/controllers/finders_controller.rb (hunk 1 above)
Comment (BeckaL) [reply app/controllers/finders_controller.rb:202]: Oh I see this method is added to in the next commit, ignore this.

**Review by BeckaL — APPROVED** [review]
Nice ✨ ! Looking good to me. Just a few very minor comments, approving anyway as non-blocking refactoring suggestions.

File: app/controllers/finders_controller.rb (hunk 2 above)
Comment (brucebolt) [reply app/controllers/finders_controller.rb]: Thanks for reviewing @BeckaL.  I've moved all of the refactoring into the first commit, so the diff is clearer between the two.

**Review by KludgeKML — APPROVED** [review]
LGTM
